### PR TITLE
General fixes

### PR DESCRIPTION
**Pull request by peppy**
- [x] This relies on all unmerged PRs at a framework level being merged.

### CODE REVIEW

File: osu.Game/GameModes/Menu/Intro.cs [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+            bgm.Looping = true;
+
+            Game.Scheduler.AddDelayed(delegate
+            {
+                bgm.Start();
+            }, 600);
+
+            Game.Scheduler.AddDelayed(delegate
+            {
+                Push(new MainMenu());
```
Comment (Tom94) [inline osu.Game/GameModes/Menu/Intro.cs]: So if I understand this correctly, Intro will always be at the bottom of the GameMode stack, correct?

File: osu.Game/OsuGame.cs [hunk 2]
```diff
@@ -34,18 +37,51 @@ public override void Load()
...
             Add(new Drawable[] {
-                MainMenu = new MainMenu(),
+                intro = new Intro(),
                 Toolbar = new Toolbar
                 {
-                    OnHome = delegate { MainMenu.MakeCurrent(); },
+                    OnHome = delegate { MainMenu?.MakeCurrent(); },
                     OnSettings = delegate { Options.PoppedOut = !Options.PoppedOut; },
                     OnPlayModeChange = delegate (PlayMode m) { PlayMode.Value = m; },
+                    Alpha = 0.001f //fixes invalidation fuckup
```
Comment (Tom94) [inline osu.Game/OsuGame.cs]: you gotta tell me more about this in next voice session

File: osu.Game/OsuGame.cs [hunk 3]
```diff
@@ -34,18 +37,51 @@ public override void Load()
...
+                Toolbar.FadeIn(100);
+
+            Cursor.FadeIn(100);
+
+            ModeChanged?.Invoke(newMode);
+        }
+
+        private void modeAdded(GameMode newMode)
+        {
+            newMode.ModePushed += modeAdded;
```
Comment (Tom94) [inline osu.Game/OsuGame.cs]: This

File: osu.Game/OsuGame.cs [hunk 4]
```diff
@@ -37,6 +37,11 @@ public override void Load()
         {
             base.Load();
 
+            //attach out bindables to the audio subsystem.
```
Comment (Tom94) [inline osu.Game/OsuGame.cs]: this happens to me sooo often, too.

**Review by Tom94 — APPROVED** [review]
Looks good to merge after you addressed the comments.

File: osu.Game/OsuGame.cs (hunk 2 above)
Comment (peppy) [inline osu.Game/OsuGame.cs]: in case i forget, you can reproduce by making this line `Alpha = 0` and adjusting the toolbar's `FadeIn` to 500ms or longer.

**Comment by peppy** [comment]
All the issues you pointed out should now be addressed!

**Pull request merged**